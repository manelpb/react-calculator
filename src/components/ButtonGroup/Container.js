import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  max-width: ${props => props.width || '154px'};
`;

export default Container;