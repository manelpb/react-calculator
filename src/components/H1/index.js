import styled from 'styled-components';

const H1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: red;
`;

export default H1;