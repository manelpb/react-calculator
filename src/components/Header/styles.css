.header {
    background-color: #222;
    height: 50px;
    padding: 10px;
    color: white;
    margin-bottom: 20px;
}