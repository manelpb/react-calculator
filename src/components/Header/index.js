import React, { PureComponent } from 'react';
import './styles.css';
import H1 from 'components/H1';
import Container from 'components/Container';

class Header extends PureComponent {
  render() {
    return (
      <Container className="header">
        <H1>React Calculator App</H1>
      </Container>
    );
  }
}

export default Header;
