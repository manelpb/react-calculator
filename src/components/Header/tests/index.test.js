import React from 'react';
import Header from '../';
import Container from 'components/Container';
import H1 from 'components/H1';
import { shallow } from 'enzyme';

it('should render a Container and H1', () => {
    const render = shallow(<Header />);
    expect(render.find(Container).length + render.find(H1).length).toEqual(2);
})