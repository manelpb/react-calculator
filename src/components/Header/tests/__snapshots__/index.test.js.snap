// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Container and H1 1`] = `
<styled.div
  className="header"
>
  <styled.h1>
    React Calculator App
  </styled.h1>
</styled.div>
`;
